### ISSUE TITLE
removeMany stops mid-remove when non-existent keys are passed

### ISSUE THREAD

**Issue by retorquere** [blinkdb-js/blinkdb#29]
I'm pretty sure I must be missing something dead obvious, but I have a script where I do two `removeMany`s that should have identical effect (I think) that don't in practice. Sorry for the newbie questions, but I've been trying to figure out all day what I'm doing wrong, and I just don't see it.

```
const { createDB , createTable, BlinkKey, insertMany, removeMany, clear, many, key } = require( "blinkdb")

const blinkdb = createDB({ clone: true })
let table = createTable(blinkdb, "users")({
  primary: "itemID",
  indexes: ["itemKey", "libraryID", "citationKey"]
})

const data = [
  {
    "citationKey": "shelahStableTheories1969",
    "itemID": 2,
    "itemKey": "6STQAJ5F",
    "libraryID": 1,
    "pinned": false
  },
  {
    "citationKey": "shelahNoteMinmaxProblem1969",
    "itemID": 4,
    "itemKey": "KG9BBSIH",
    "libraryID": 1,
    "pinned": false
  },
]

const deletes = [
  { "itemID": 2 },
  { "itemID": 5 },
  { "itemID": 4 },
]

async function main() {
  await insertMany(table, data)

  // this works partially
  await removeMany(table, deletes)
  console.log('remaining:', (await many(table)).length)
  console.log('ghosts:', (await many(table, { where: { itemID: { in: deletes.map(d => d.itemID) } } })).length)

  await clear(table)
  await insertMany(table, data)
  // this does work
  deletes.sort((a, b) => a.itemID - b.itemID)
  await removeMany(table, deletes)
  console.log('remaining:', (await many(table)).length)
  console.log('ghosts:', (await many(table, { where: { itemID: { in: deletes.map(d => d.itemID) } } })).length)
}

main().catch(err => console.log(err))
```

**Comment by retorquere** [comment]
Trimming the deletes to the effective deletes does work. Does `removeMany` not tolerate non-existent keys to be passed?

**Comment by retorquere** [comment]
Looks like [it doesn't](https://github.com/blinkdb-js/blinkdb/blob/main/packages/db/src/core/removeMany.ts#L45). I had not expected this behavior. Does that mean I should always run a `many` query to detect existing items before passing ids to `removeMany`? This seems fairly counterintuitive to me.

**Comment by retorquere** [comment]
Worse -- it does a partial delete until it finds an ID it doesn't know and then stops (I've changed the script to show this, but the code I linked to makes it clear it will). Shouldn't that `return false` just be a `continue`?

**Comment by froehlichA** [comment]
Yep, that's an error. Deleting an ID that does not exist within the table should be a noop.

Would you mind making a PR for this? :)

**Comment by retorquere** [comment]
https://github.com/blinkdb-js/blinkdb/pull/30

**Comment by retorquere** [comment]
Any indication when this could be out in a new release? I'm mid-migration to blinkdb in a tool I'm making and this is a showstopper for me.

**Comment by froehlichA** [comment]
closed in #30.

**Comment by froehlichA** [comment]
> Any indication when this could be out in a new release? I'm mid-migration to blinkdb in a tool I'm making and this is a showstopper for me.

See https://github.com/blinkdb-js/blinkdb/releases/tag/v0.14.0.
